From: Dave Airlie <airlied@redhat.com>
Date: Thu, 27 Mar 2008 05:18:39 +0000 (+1000)
Subject: quirk: fix LPL monitors properly.
X-Git-Url: http://gitweb.freedesktop.org/?p=xorg/xserver.git;a=commitdiff;h=c747030a49dd289e873e2b686cd129d840e55468

quirk: fix LPL monitors properly.

no point having a h cm fix when we really want to copy the sizes from the
other place.

RH BZ 435216
---

[Modified by Bryce Harrington to apply to Ubuntu]

diff -Nurp xorg-server-1.4.1~git20080131-patched/hw/xfree86/modes/xf86EdidModes.c xorg-server-1.4.1~git20080131-working/hw/xfree86/modes/xf86EdidModes.c
--- xorg-server-1.4.1~git20080131-patched/hw/xfree86/modes/xf86EdidModes.c	2008-05-12 17:52:40.000000000 -0700
+++ xorg-server-1.4.1~git20080131-working/hw/xfree86/modes/xf86EdidModes.c	2008-05-12 23:07:43.000000000 -0700
@@ -104,17 +104,6 @@ static Bool quirk_prefer_large_75 (int s
 
 static Bool quirk_detailed_h_in_cm (int scrnIndex, xf86MonPtr DDC)
 {
-    /* Bug #10304: "LGPhilipsLCD LP154W01-A5" */
-    /* Bug #12784: "LGPhilipsLCD LP154W01-TLA2" */
-    if (memcmp (DDC->vendor.name, "LPL", 4) == 0 &&
-	DDC->vendor.prod_id == 0)
-	return TRUE;
-
-    /* LP Bug #127827; "LGPhilipsLCD LP154W01-TLAE" */
-    if (memcmp (DDC->vendor.name, "LPL", 4) == 0 &&
-        DDC->vendor.prod_id == 10752)
-        return TRUE;
-
     /* Bug #11603: Funai Electronics PM36B */
     if (memcmp (DDC->vendor.name, "FCM", 4) == 0 &&
 	DDC->vendor.prod_id == 13600)
@@ -136,8 +125,9 @@ static Bool quirk_detailed_v_in_cm (int 
 static Bool quirk_detailed_use_maximum_size (int scrnIndex, xf86MonPtr DDC)
 {
     /* Bug #10304: LGPhilipsLCD LP154W01-A5 */
+    /* LP Bug #127827; "LGPhilipsLCD LP154W01-TLAE" */
     if (memcmp (DDC->vendor.name, "LPL", 4) == 0 &&
-	DDC->vendor.prod_id == 0)
+	(DDC->vendor.prod_id == 0 || DDC->vendor.prod_id == 0x2a00))
 	return TRUE;
 
     return FALSE;
